c/zssLogging.c

```diff
@@ -43,6 +45,182 @@ bool isLogLevelValid(int level) {
   return TRUE;
 }
 
+typedef struct LogMessagePrefix_tag {
+  char text[LOG_MSG_PREFIX_SIZE];
+} LogMessagePrefix;
+
+typedef struct LogTimestamp_tag {
+  union {
+    struct {
+      char year[4];
+      char delim00[1];
+      char month[2];
+      char delim01[1];
+      char day[2];
+      char delim02[1];
+      char hour[2];
+      char delim03[1];
+      char minute[2];
+      char delim04[1];
+      char second[2];
+      char delim05[1];
+      char hundredth[2];
+    } dateAndTime;
+    char text[22];
+  };
+} LogTimestamp;
+
+
+static void getCurrentLogTimestamp(LogTimestamp *timestamp) {
+
+  uint64 stck = 0;
+  getSTCKU(&stck);
+
+  stck += getLocalTimeOffset();
+
+  stckToLogTimestamp(stck, timestamp);
+
+}
+
+static void getLocationData(char *path, int line, char **locationInfo, uint64 compID, LoggingComponent *component) {
+
+  char prefix[PREFIX_SUFFIX_SIZE]; 
+  char suffix[PREFIX_SUFFIX_SIZE]; 
+
+  unsigned int id = compID & 0xFFFFFFFF;
+  if(compID >= LOG_PROD_COMMON && compID < LOG_PROD_ZIS) {
+    snprintf(prefix,sizeof(LOG_PREFIX_ZCC),LOG_PREFIX_ZCC);
+    switch (id) { // most of these don't actally log, but are defined in logging.h
```

Comment:
Can you please use the switch statement format we've already used in the ZSS code? 

For example,
```c
static int translateLogLevelToClientTraceLevel(int logLevel) {

  int traceLevel = CLIENT_TRACE_LEVEL_NA;
  switch (logLevel) {
  case ZOWE_LOG_SEVERE:
    traceLevel = CLIENT_TRACE_LEVEL_SEVERE;
    break;
  case ZOWE_LOG_WARNING:
    traceLevel = CLIENT_TRACE_LEVEL_WARNING;
    break;
  case ZOWE_LOG_INFO:
    traceLevel = CLIENT_TRACE_LEVEL_INFO;
    break;
  case ZOWE_LOG_DEBUG:
    traceLevel = CLIENT_TRACE_LEVEL_FINE;
    break;
  case ZOWE_LOG_DEBUG2:
    traceLevel = CLIENT_TRACE_LEVEL_FINER;
    break;
  case ZOWE_LOG_DEBUG3:
    traceLevel = CLIENT_TRACE_LEVEL_FINEST;
    break;
  }

  return traceLevel;
}
```